operator/pkg/ciliumidentity/controller_test.go

```diff
@@ -414,31 +405,27 @@ func TestUpdateUsedCIDIsReverted(t *testing.T) {
 		t.Fatalf("update CID: %v", err)
 	}
 
-	// Wait for update event to propagate.
-	ev := <-events
-	if ev.Kind != resource.Upsert {
-		t.Fatalf("expected upsert event, got %v", ev.Kind)
-	}
-	if ev.Key.Name != toUpdate.Name {
-		t.Fatalf("expected upsert event for cid %s, got %v", toUpdate.Name, ev.Key.Name)
+	cid, err := fakeClient.CiliumV2().CiliumIdentities().Get(ctx, updated.Name, metav1.GetOptions{})
```

Comment:
Isn't it possible that CID controller would revert the CID before the test gets it here?

I assume you ran the stress test and it didn't fail.